tests/aws/services/sqs/test_sqs_backdoor.py

```diff
@@ -361,3 +364,128 @@ def test_list_messages_with_queue_url_in_path(
+                params["headers"][HEADER_LOCALSTACK_SQS_OVERRIDE_MESSAGE_COUNT] = (
+                    override_message_count
+                )
+
+        aws_client.meta.events.register(
+            "provide-client-params.sqs.ReceiveMessage", _handle_receive_message_override
+        )
+
+        aws_client.meta.events.register("before-call.sqs.ReceiveMessage", _handler_inject_header)
+
+        response = aws_client.receive_message(
+            QueueUrl=queue_url,
+            VisibilityTimeout=30,
+            MaxNumberOfMessages=override_max_number_of_messages,
+            AttributeNames=["All"],
+        )
+
+        messages = response.get("Messages", [])
+        assert len(messages) == 20
+
+    @markers.aws.only_localstack
+    def test_receive_message_override_message_wait_time_seconds(
+        self, sqs_create_queue, aws_client_factory
+    ):
+        aws_client = aws_client_factory().sqs
+
+        override_message_wait_time_seconds = 30
+
+        from localstack.services.sqs.constants import (
+            HEADER_LOCALSTACK_SQS_OVERRIDE_WAIT_TIME_SECONDS,
+        )
+        from localstack.services.sqs.provider import MAX_NUMBER_OF_MESSAGES
+
+        queue_url = sqs_create_queue()
+
+        with pytest.raises(ClientError):
+            aws_client.receive_message(
+                QueueUrl=queue_url,
+                VisibilityTimeout=0,
+                MaxNumberOfMessages=MAX_NUMBER_OF_MESSAGES,
+                WaitTimeSeconds=override_message_wait_time_seconds,
+                AttributeNames=["All"],
+            )
+
+        def _handle_receive_message_override(params, context, **kwargs):
+            if not (requested_wait_time := params.get("WaitTimeSeconds")):
+                return
+            context[HEADER_LOCALSTACK_SQS_OVERRIDE_WAIT_TIME_SECONDS] = str(requested_wait_time)
+            params["WaitTimeSeconds"] = min(20, requested_wait_time)
+
+        def _handler_inject_header(params, context, **kwargs):
+            if override_wait_time := context.pop(
+                HEADER_LOCALSTACK_SQS_OVERRIDE_WAIT_TIME_SECONDS, None
+            ):
+                params["headers"][HEADER_LOCALSTACK_SQS_OVERRIDE_WAIT_TIME_SECONDS] = (
+                    override_wait_time
+                )
+
+        aws_client.meta.events.register(
+            "provide-client-params.sqs.ReceiveMessage", _handle_receive_message_override
+        )
+
+        aws_client.meta.events.register("before-call.sqs.ReceiveMessage", _handler_inject_header)
+
+        def _send_message():
+            aws_client.send_message(QueueUrl=queue_url, MessageBody=f"message-{short_uid()}")
+
+        # Populate with 9 messages (1 below the MaxNumberOfMessages threshold).
+        # This should cause long-polling to exit since MaxNumberOfMessages is met.
+        for _ in range(9):
+            _send_message()
+
+        Timer(25, _send_message).start()  # send message asynchronously after 1 second
+
+        start_t = time.time()
```

Comment:
We should use `time.perf_counter` here, as `time.time` is not guaranteed to be monotonic. It is quite unlikely, but it could result in less than 25 seconds measured, if the system clock is altered in that timeframe.